extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper} from '../../../src/preact/component';
+import {assertDoesNotContainDisplay, setStyles} from '../../../src/style';
+import {forwardRef} from '../../../src/preact/compat';
+import {
+  useCallback,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from '../../../src/preact';
+import {useStyles} from './sidebar.jss';
+
+const ANIMATION_DURATION = 350;
+const ANIMATION_EASE_IN = 'cubic-bezier(0,0,.21,1)';
+
+const ANIMATION_KEYFRAMES_FADE_IN = [{'opacity': '0'}, {'opacity': '1'}];
+const ANIMATION_KEYFRAMES_SLIDE_IN_LEFT = [
+  {'transform': 'translateX(-100%)'},
+  {'transform': 'translateX(0)'},
+];
+const ANIMATION_KEYFRAMES_SLIDE_IN_RIGHT = [
+  {'transform': 'translateX(100%)'},
+  {'transform': 'translateX(0)'},
+];
+
+const ANIMATION_STYLES_SIDEBAR_LEFT_INIT = {'transform': 'translateX(-100%)'};
+const ANIMATION_STYLES_SIDEBAR_RIGHT_INIT = {'transform': 'translateX(100%)'};
+const ANIMATION_STYLES_MASK_INIT = {'opacity': '0'};
+const ANIMATION_STYLES_SIDEBAR_FINAL = {'transform': ''};
+const ANIMATION_STYLES_MASK_FINAL = {'opacity': '1'};
+
+/**
+ * @param {T} current
+ * @return {{current: T}}
+ * @template T
+ */
+function useValueRef(current) {
+  const valueRef = useRef(null);
+  valueRef.current = current;
+  return valueRef;
+}
+/**
+ * @param {!Element} element
+ * @param {!Object<string, *>} styles
+ */
+function safelySetStyles(element, styles) {
```

Comment:
Why is this needed?

Reply:
So the thought process is like this:

In order to have styles defined in the module level constants, and not directly inline in `setStyles` then we have to include the `assertDoesNotContainDisplay` wrapper around the styles object that is passed in.  i.e. 
```
setStyles(element, {opacity: 1}) //okay

const style = {opacity:1}
setStyles(element, style) //not okay
setStyles(element, assertDoesNotContainDisplay(style)) //okay
```

So I thought it made sense to include the wrapper function `safelySetStyles` to clean up the syntax where ever we use `setStyles`

What do you think?  We can also put the `assertDoesNotContainDisplay` directly inline, but it looks a lot less clean in my opinion.

Reply:
Ok. I see. I suppose this is ok then. Obviously seems weird since the styles are right there. But I suppose if tooling forces us into this then it's ok.

Reply:
Sounds good!

Reply:
Should `safelySetStyles` be moved to `style.js`?